server/test/articles.spec.js

```diff
@@ -309,4 +317,143 @@ describe('Article endpoints', () => {
         });
     });
   });
+
+  describe('The endpoint for article search', () => {
+    describe('Filter by article title', () => {
+      it('should search article by title', (done) => {
+        chai.request(app)
+          .get(`/api/search?title=${title}`)
+          .end((error, res) => {
+            expect(res.status).to.be.equal(200);
+            expect(res.body).to.have.property('data');
+            expect(res.body.data).to.have.property('articles');
+            done();
+          });
+      });
+
+      it('should fail if title not provided', (done) => {
+        chai.request(app)
+          .get('/api/search?title=')
+          .end((error, res) => {
+            expect(res.status).to.be.equal(400);
+            expect(res.body).to.have.property('message');
+            expect(res.body.message).to.have.equals('title is not allowed to be empty');
+            done();
+          });
+      });
+    });
+
+    describe('Filter by article author', () => {
+      it('should search article by author', (done) => {
+        chai.request(app)
+          .get(`/api/search?author=${author}`)
+          .end((error, res) => {
+            expect(res.status).to.be.equal(200);
+            expect(res.body).to.have.property('data');
+            expect(res.body.data).to.have.property('articles');
+            done();
+          });
+      });
+
+      it('should fail if author not provided', (done) => {
+        chai.request(app)
+          .get('/api/search?author=')
+          .end((error, res) => {
+            expect(res.status).to.be.equal(400);
+            expect(res.body).to.have.property('message');
+            expect(res.body.message).to.have.equals('author is not allowed to be empty');
+            done();
+          });
+      });
+    });
+
```

Comment:
I would suggest testing also if the author does not exist, or tag or article title.

Reply:
> I would suggest testing also if the author does not exist, or tag or article title.

@dusmel. If the user input an author who doesn't exist, at least we are searching author with a name which `looks like` it. In case no name looks like the author name. We are displaying an empty array. For the  `empty array` the solution is here https://github.com/andela/ah-kgl-wakanda-backend/pull/30#discussion_r281453906